tephra-hbase-compat-1.1-base/src/main/java/org/apache/tephra/hbase/coprocessor/janitor/HBaseTransactionPruningPlugin.java

```diff
@@ -259,6 +262,8 @@ private long computePruneUpperBound(TimeRegions timeRegions) throws IOException
         if (pruneUpperBoundForTime != -1) {
           Long minPruneUpperBoundRegions = Collections.min(pruneUpperBoundRegions.values());
           return Math.min(pruneUpperBoundForTime, minPruneUpperBoundRegions);
+        } else {
+          LOG.debug("Ignoring invalid prune upper bound -1 for time {}", time);
         }
       } else {
         if (LOG.isDebugEnabled()) {
```

Comment:
Maybe the message here should also be rephrased: "Ignoring regions for time {} because not all of the regions recorded a pruneUpperBound."

This may seem picky, but remember that these messages will be seen by devops who does not understand the internals. If you rephrase it this way, they will understand better what is gong on.